addons/mail/static/src/discuss/core/common/poll.js

```diff
@@ -0,0 +1,94 @@
+import { Component, onMounted, onWillUnmount, useEffect, useState } from "@odoo/owl";
+
+import { _t } from "@web/core/l10n/translation";
+
+/**
+ * @typedef {Object} Props
+ * @property {import("models").Poll} poll
+ * @extends {Component<Props, Env>}
+ */
+export class Poll extends Component {
+    static template = "mail.Poll";
+    static props = ["poll"];
+
+    setup() {
+        this.state = useState({
+            remainingTimeText: "",
+            showResults: false,
+            selectedOptionIds: new Set(),
+            voting: false,
+        });
+        useEffect(
+            (alreadyVoted, endMessageId) => {
+                this.state.showResults = alreadyVoted || endMessageId;
+            },
+            () => [this.props.poll.selfAlreadyVoted, this.props.poll.end_message_id]
+        );
+        let remainingTimeInterval;
+        const updateRemainingTime = () => {
+            const diff = this.props.poll.endDateTime.diffNow(["hours", "minutes"]);
+            if (diff.valueOf() <= 0) {
+                this.state.remainingTimeText = _t("Poll will end soon");
+                clearInterval(remainingTimeInterval);
+                return;
+            }
+            const unit = diff.as("minutes") < 60 ? "minutes" : "hours";
+            this.state.remainingTimeText = _t(
+                "%s left",
+                diff.shiftTo(unit).toHuman({ maximumFractionDigits: 0 })
+            );
+        };
+        onMounted(() => {
+            updateRemainingTime();
+            remainingTimeInterval = setInterval(updateRemainingTime, 60_000);
+        });
+        onWillUnmount(() => clearInterval(remainingTimeInterval));
+    }
+
+    toggleShowResults() {
+        this.state.showResults = !this.state.showResults;
+    }
+
+    async onClickVote() {
+        try {
+            this.state.voting = true;
+            await this.props.poll.vote([...this.state.selectedOptionIds]);
+        } finally {
+            this.state.voting = false;
+            this.state.selectedOptionIds.clear();
+        }
+    }
+
+    onOptionCheckboxToggle(optionId, checked) {
+        if (!this.props.poll.allow_multiple_options) {
+            this.state.selectedOptionIds.clear();
+        }
+        if (checked) {
+            this.state.selectedOptionIds.add(optionId);
+        } else {
+            this.state.selectedOptionIds.delete(optionId);
+        }
+    }
+
+    get voteButtonDisabled() {
+        return this.state.selectedOptionIds.size === 0 || this.state.voting;
+    }
+
+    percentageAttStyle(option) {
+        return this.state.showResults ? `background-size: ${option.vote_percentage}%;` : "";
+    }
+
+    optionVoteCountText(option) {
+        if (option.number_of_votes === 1) {
+            return _t("%(count)s vote", { count: option.number_of_votes });
```

Comment:
No need for `count`: this is necessarily "1".

Reply:
:face_with_peeking_eye: